src/canvas-view/canvas/node-toolbar.component.js

```diff
@@ -0,0 +1,32 @@
+(function () {
+  'use strict';
+
+  angular.module('patternfly.canvas')
+    .component('nodeToolbar', {
+      templateUrl: 'canvas-view/canvas/node-toolbar.html',
+      bindings: {
+        node: '=',
+        nodeActions: '=',
+        nodeClickHandler: '=',
+        nodeCloseHandler: '='
+      },
+      controller: function NodeToolbarController () {
+        var ctrl = this;
+        ctrl.selectedAction = "none";
+
+        ctrl.actionIconClicked = function (action) {
+          ctrl.selectedAction = action;
+          if (angular.isFunction(ctrl.nodeClickHandler)) {
+            ctrl.nodeClickHandler({'action': action, 'node': ctrl.node});
```

Comment:
Might be better to pass 2 arguments rather than an object.

Reply:
I agree